server/app.py

```diff
@@ -144,9 +153,13 @@ def add_suffix_to_name(stop):
 
 def helper_thread():
     while True:
-        update_cache()
-        logging.debug("helper thread updated cache with predictions")
-        MetricsHandler.cache_last_updated.set(int(time.time()))
+        try:
+            update_cache()
+            logging.debug("helper thread updated cache with predictions")
+            MetricsHandler.cache_last_updated.set(int(time.time()))
+        except Exception as e:
+            logging.exception(f"Exception in helper_thread: {e}")
```

Comment:
`logging.exception` will automatically include the error in the log
```suggestion
        except Exception:
            logging.exception("Unable to update cache")
```